Update Castle destination docs (#3610)
segmentio/segment-docs: src/connections/destinations/catalog/castle/index.md

title: Castle Destination
id: 56a8f566e954a874ca44d3b0
---
[Castle](https://castle.io/?utm_source=segmentio&utm_medium=docs&utm_campaign=partners){:target="_blank"} monitors every step of the customer journey to help visualize and proactively block fraud that would otherwise fly under the radar. Types of fraud or abuse that can be managed include bots, fake accounts, multi-accounting, and account sharing.

> This destination is maintained by Castle.

Author: ```suggestion
Castle maintains this destination. For any issues with the destination, contact the [Castle support team](mailto:support@castle.io).
```

## Getting Started

1. From the Segment web app, click **Catalog**.
2. Search for "Castle" in the Catalog, select it, and choose which of your sources to connect the destination to.

> 3. Enter the "Publishable Key" into your Segment Settings UI which you can find in your property in the [Castle dashboard](https://dashboard.castle.io).

Reviewer: ```suggestion
3. Enter the "Publishable Key" the Publishable Key field. Find the Publishable Key on the Castle dashboard.
```

4. Calls will now be visible in Castle dashboards in real-time.

> **NOTE**: Castle will only ingest Segment _client-side_ events at this point. Server-side events will be dropped and not processed.

Reviewer: ```suggestion
> info ""
> Castle ingests Segment Client-side events. Server-side events are dropped and not processed.
> Castle supports Segment web integrations.
```

Add info callout, use active voice, consolidate with note below.

> **NOTE**: Castle will only ingest Segment _client-side_ events at this point. Server-side events will be dropped and not processed.

Reviewer 2: ```suggestion
> warning ""
> Castle only ingests Segment _client-side_ events at this point. Server-side events are dropped and won't be processed.
```

> **NOTE**: Castle will only ingest Segment _client-side_ events at this point. Server-side events will be dropped and not processed.

Author: ```suggestion
> info ""
> Castle ingests Segment Client-side events. Server-side events are dropped and not processed.
> Castle only supports web integrations through Segment, but is in the process of working on mobile support.
```

> **NOTE**: Castle only supports web integrations via Segment, but we are working on mobile support so please stay tuned.

Reviewer 2: ```suggestion
> info ""
> Castle only supports web integrations through Segment, but is in the process of working on mobile support.
```

> **NOTE**: Castle only supports web integrations via Segment, but we are working on mobile support so please stay tuned.

Reviewer: ```suggestion

```

Remove this line and add to info callout above.

> ***

Reviewer: ```suggestion
```

## Page

> If you're not familiar with the Segment Specs, take a look to understand what the [`page` method](https://segment.com/docs/connections/spec/page/) does. An example call would look like:

Reviewer 2: ```suggestion
Take a look to understand what the [Page method](/docs/connections/spec/page/) does. An example call looks like:
```

> If you're not familiar with the Segment Specs, take a look to understand what the [`page` method](https://segment.com/docs/connections/spec/page/) does. An example call would look like:

Reviewer: ```suggestion
If you're not familiar with the Segment Specs, take a look to understand what the [Page call](/docs/connections/spec/page/) does. An example call looks like:
```

Relative URLs for other docs pages

```
analytics.page()
```

> `page` calls will be sent to Castle as `$page` events.

Reviewer 2: ```suggestion
`page` send to Castle as `$page` events.
```

> `page` calls will be sent to Castle as `$page` events.

Reviewer: ```suggestion
Segment sends Page calls to Castle as `$page` events.
```
Formatting for Segment methods / calls, and active voice.

## Track

> If you're not familiar with the Segment Specs, take a look to understand what the [Track method](https://segment.com/docs/connections/spec/track/) does. An example call would look like:

Reviewer: ```suggestion
If you're not familiar with the Segment Specs, take a look to understand what the [Track method](/docs/connections/spec/track/) does. An example call looks like:
```

```
analytics.track('Added to Cart')
```

> `track` calls will be sent to Castle as a `$custom` events.

Reviewer 2: ```suggestion
`track` calls send to Castle as a `$custom` events.
```

> `track` calls will be sent to Castle as a `$custom` events.

Reviewer: ```suggestion
Segment sends Track calls to Castle as a `$custom` events.
```

***

## Secure Mode

> In order to prevent user information from being spoofed by a bad actor, it is highly recommended to send the user information as a signed JWT when Castle.js is used in production.

Reviewer 2: ```suggestion
In order to prevent user information from being spoofed by a bad actor, Castle recommends you to send the user information as a signed JWT when Castle.js is used in production.
```

> In order to prevent user information from being spoofed by a bad actor, it is highly recommended to send the user information as a signed JWT when Castle.js is used in production.

Reviewer: ```suggestion
Send user information as a signed JWT when you use Castle in production. This prevents bad actors from spoofing any user information.
```
Simplify the wording here.

> From your backend code, you need to encode the user as a JWT and sign it using your Castle "API Secret". Then, when Castle receives the JWT, the integrity of the user data will be verified to ensure that the data isn't being tampered with.

Reviewer 2: ```suggestion
From your backend code, you need to encode the user as a JWT and sign it using your Castle **API Secret**. Then, when Castle receives the JWT, the integrity of the user data will be verified to ensure that the data isn't being tampered with.
```

> From your backend code, you need to encode the user as a JWT and sign it using your Castle "API Secret". Then, when Castle receives the JWT, the integrity of the user data will be verified to ensure that the data isn't being tampered with.

Reviewer: ```suggestion
In your backend code, encode the user as a JWT and sign it using your Castle "API Secret". When Castle receives the JWT, the integrity of the user data is verified to ensure that the data wasn't tampered with.
```

Below is an example of how to generate a JWT on your backend using the Ruby language:

```ruby
jwt_from_backend = JWT.encode({
  id: '97980cfea0067',
  email: 'peter@example.com'
}, ENV.fetch('CASTLE_API_SECRET'), 'HS256')
```

> You then need to transfer the `user_jwt` object to your frontend either via a separate API call, or by injecting the code using a templating language:

Reviewer: ```suggestion
Transfer the `user_jwt` object to your frontend through a separate API call, or by injecting the code using a templating language:
```

```javascript
var userJwt = "<%= jwt_from_backend %>";

// Then use the `userJwt` argument instead of `user` when using any of the tracking methods
Castle.page({userJwt: userJwt});

analytics.identify('97980cfea0067', {
  email: 'peter@example.com',
}, {
  Castle: {
    userJwt: userJwt
  }
});
```

> When Castle receives a JWT version of the user object, its contents will override the user object sent the standard Segment way.

Reviewer 2: ```suggestion
When Castle receives a JWT version of the user object, its contents override the user object sent the standard Segment way.
```

> When Castle receives a JWT version of the user object, its contents will override the user object sent the standard Segment way.

Reviewer: ```suggestion
When Castle receives a JWT version of the user object, its contents override the user object sent the standard Segment way.
```
